src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java

```diff
@@ -0,0 +1,159 @@
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.ApplicationParams;
+import uk.gov.hmcts.ccd.data.casedetails.search.MetaData;
+import uk.gov.hmcts.ccd.data.casedetails.search.SortOrderField;
+import uk.gov.hmcts.ccd.data.user.UserService;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition;
+import uk.gov.hmcts.ccd.domain.service.aggregated.AuthorisedGetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.GetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.SearchQueryOperation;
+import uk.gov.hmcts.ccd.domain.service.common.ObjectMapperService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadSearchRequest;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+@Service
+@Slf4j
+public class ElasticsearchQueryHelper {
+
+    private static final String SORT = "sort";
+    private static final String CASE_DATA_PREFIX = "data.";
+    private static final String COLLECTION_VALUE_SUFFIX = ".value";
+    private static final String KEYWORD_SUFFIX = ".keyword";
+
+    private final ObjectMapper objectMapper;
+    private final ApplicationParams applicationParams;
+    private final ObjectMapperService objectMapperService;
+    private final SearchQueryOperation searchQueryOperation;
+    private final GetCaseTypeOperation getCaseTypeOperation;
+    private final UserService userService;
+    private final ElasticsearchMappings elasticsearchMappings;
+
+    @Autowired
+    public ElasticsearchQueryHelper(@Qualifier("DefaultObjectMapper") ObjectMapper objectMapper,
+                                    ApplicationParams applicationParams,
+                                    ObjectMapperService objectMapperService,
+                                    SearchQueryOperation searchQueryOperation,
+                                    @Qualifier(AuthorisedGetCaseTypeOperation.QUALIFIER) GetCaseTypeOperation getCaseTypeOperation,
+                                    UserService userService,
+                                    ElasticsearchMappings elasticsearchMappings) {
+        this.objectMapper = objectMapper;
+        this.applicationParams = applicationParams;
+        this.objectMapperService = objectMapperService;
+        this.searchQueryOperation = searchQueryOperation;
+        this.getCaseTypeOperation = getCaseTypeOperation;
+        this.userService = userService;
+        this.elasticsearchMappings = elasticsearchMappings;
+    }
+
+    public CrossCaseTypeSearchRequest prepareRequest(List<String> caseTypeIds, String useCase, String jsonSearchRequest) {
```

Comment:
some elements are not suitable for the external API. for example the concept of use case, and the adding of the default sorting. If we want to keep this common class shared between the external and internal api, then let's keep only the things strictly in common